### PR TITLE
fix the issue that the output dict of jit model could not get [:2]

### PR DESCRIPTION
**Pull request by sywangyi**
"TypeError: unhashable type: 'slice'"

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

- pipelines: @Narsil

### CODE REVIEW

**Comment by sywangyi** [comment]
@yao-matrix

**Review by Narsil — APPROVED** [review]
Works for me. 

@sgugger

File: src/transformers/pipelines/question_answering.py [hunk 1]
```diff
@@ -509,8 +509,8 @@ def preprocess(self, example, padding="do_not_pad", doc_stride=None, max_questio
     def _forward(self, inputs):
         example = inputs["example"]
         model_inputs = {k: inputs[k] for k in self.tokenizer.model_input_names}
-        start, end = self.model(**model_inputs)[:2]
```
Comment (sgugger) [inline src/transformers/pipelines/question_answering.py:512]: Model outputs can either be regular tuples or our hybrid subclass of dicts that also support slice indexing. So here there should be a test on `outputs = self.model(**model_inputs)`. If it satisfies `isinstance(outputs, tuple)` we should leave the code as is, and in the else branch we can use your changes.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this! Left a comment.

File: src/transformers/pipelines/question_answering.py (hunk 1 above)
Comment (sywangyi) [reply src/transformers/pipelines/question_answering.py:512]: thanks for the comments. fixed

**Review by sgugger — APPROVED** [review]
Thanks for iterating on this!

**Pull request merged**